src/main/java/com/mturko/bestshop/model/AbstractEntity.java

```diff
@@ -0,0 +1,20 @@
+package com.mturko.bestshop.model;
+
+import java.util.Objects;
+
+public class AbstractEntity {
+    private int id;
```

Comment:
Are you sure that 2 bilions is enough max value for identifiers in your store? :-) I would suggest to use long for this purpose. See for details section "4.2.1. Integral Types and Values
" on the following page: https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.2

And also please use wrapper tipe because this property can have `null` value.